src/OrchardCore/OrchardCore.Mvc.Core/RazorPages/PageConventionCollectionExtensions.cs

```diff
@@ -17,13 +17,14 @@ public static class PageConventionCollectionExtensions
             {
                 foreach (var selector in model.Selectors.ToArray())
                 {
-                    if (selector.AttributeRouteModel.Template.StartsWith(areaName, StringComparison.Ordinal))
+                    var path = $"{areaName}/{folderPath.Trim('/')}";
+                    if (selector.AttributeRouteModel.Template.Contains( path, StringComparison.Ordinal))
                     {
                         selector.AttributeRouteModel.SuppressLinkGeneration = true;
 
-                        var template = (folderRoute.Trim('/') + '/' + selector.AttributeRouteModel
-                            .Template.Substring(areaName.Length).TrimStart('/')).TrimEnd('/');
-
+                        var templatePart = selector.AttributeRouteModel.Template.Split(path);                        
+                        var template = templatePart[0].TrimEnd('/') + '/' + folderRoute.Trim('/') + '/' + templatePart[1].TrimStart('/');                        
```

Comment:
So here it seems that the template will still start with the `area + folderPath` segments, then `folderRoute` segments, then the remaining segments of the origonal template, i think this is wrong, it should be `folderRoute + remaining segments` without `area + folderPath` as done and tried with my code, so maybe better to just keep my code ;)

Note: Normally we avoid overuse of string interpollations in hot paths, maybe also to allocate an array with `Split()`, but here in a convention that runs once that's okay.

Hmm maybe good to use `PathString` here because of its + operator that take care of double slashes, but `PathString` has its traps. Above you used `AttributeRouteModel.CombineTemplates`, is it an mvc extension? Maybe useful here. But that's okay to keep it simple with string operations.

Reply:
I guess CombineTemplates takes care of '/'

> : Normally we avoid overuse of string interpollations in hot paths

Yes, I prefer `$"{X}/{Y}"`, however here I have tried to stay aligned with your code. 

> think this is wrong, it should be `folderRoute + remaining segments` without `area + folderPath` as done and tried with my code, so maybe better to just keep my code ;)

Your code worked until I added, 
`public class AdminFolderModelConvention : IPageRouteModelConvention`

Template no longer starting with `areaName + folderPath ` now it starts with `AdminPrefixUrl + areaName + folderPath `

Here I am trying to replace `areaName + folderPath ` with  `folderRoute ` but not used `String.Replace` to take care of `/`, If its Admin route then, `areaName + folderPath ` will be somewhere in the middle not at start of the template.